src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -0,0 +1,429 @@
+import _ from 'lodash/fp';
+import { Fragment, useState } from 'react';
+import { div, h, h2, h3, strong } from 'react-hyperscript-helpers';
+import { ButtonOutline, ButtonPrimary, Link, Select } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { DatasetResponse } from 'src/libs/ajax/DatasetBuilder';
+import colors from 'src/libs/colors';
+import * as Utils from 'src/libs/utils';
+import {
+  Cohort,
+  Criteria,
+  CriteriaGroup,
+  DatasetBuilderState,
+  DomainCriteria,
+  DomainType,
+  newCriteriaGroup,
+  ProgramDataListCriteria,
+  ProgramDataListType,
+  ProgramDataRangeCriteria,
+  ProgramDataRangeType,
+} from 'src/pages/library/datasetBuilder/dataset-builder-types';
+import { HomepageState, OnStateChangeType } from 'src/pages/library/datasetBuilder/DatasetBuilder';
+import { datasetBuilderCohorts } from 'src/pages/library/datasetBuilder/state';
+
+const PAGE_PADDING_HEIGHT = 0;
+const PAGE_PADDING_WIDTH = 3;
+
+const renderCriteria = (deleteCriteria: (criteria) => void) => (criteria: Criteria) =>
+  div(
+    {
+      style: {
+        display: 'flex',
+        width: '100%',
+        justifyContent: 'space-between',
+        alignItems: 'baseline',
+        paddingBottom: 5,
+        marginTop: 10,
+        marginBottom: 5,
+        borderBottom: `1px solid ${colors.dark(0.35)}`,
+      },
+    },
+    [
+      div({ style: { margin: '5px', display: 'flex', alignItems: 'center' } }, [
+        h(
+          Link,
+          {
+            'aria-label': 'delete criteria',
+            onClick: () => {
+              deleteCriteria(criteria);
+            },
+          },
+          [icon('minus-circle', { size: 24, style: { color: colors.danger() } })]
+        ),
+        div({ style: { marginLeft: 5 } }, [
+          Utils.cond(
+            [
+              'category' in criteria,
```

Comment:
Flagging this as something that Miguel might have another way to do for our meeting on friday.